src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -878,9 +1069,47 @@ private static SourceLocation GetCallerLocation(SyntaxNode syntax, ThreeState en
         /// we will provide caller information as a value of this parameter.
         /// This is done to match the native compiler behavior and user requests (see http://roslyn.codeplex.com/workitem/171). This behavior
         /// does not match the C# spec that currently requires to provide caller information only in explicit invocations and query expressions.
-        /// </remarks>
+        /// </remarks>  
         private BoundExpression GetDefaultParameterValue(SyntaxNode syntax, ParameterSymbol parameter, ThreeState enableCallerInfo)
         {
+            return GetDefaultParameterValue(syntax, parameter, enableCallerInfo, this, null, null);
+        }
+
+        /// <summary>
+        /// This helper is used by both LocalRewriter and IOperation. 
+        ///   - For lowering, 'localRewriter' must be passed in as an argument, and set 'binder' and 'diagnostics' to null.
+        ///   - For deriving argument expression for IArgument operation, 'localRewriter' must be null, and 'compilation', 'diagnostics' 
+        ///     must be passed in, where 'callerMemberName' must not be null if 'parameter.IsCallerMemberName' is 'true'.
+        /// </summary>
+        private static BoundExpression GetDefaultParameterValue(
+            SyntaxNode syntax, 
+            ParameterSymbol parameter,
+            ThreeState enableCallerInfo,
+            LocalRewriter localRewriter,
+            Binder binder,
+            DiagnosticBag diagnostics)
+        {
+            Debug.Assert(localRewriter == null ^ (binder == null && diagnostics == null));
+
+            bool isLowering;
+            bool inExpressionLambda = false;
+            MethodSymbol currentMethod = null;
+            CSharpCompilation compilation; 
+
+            if (localRewriter != null)
+            {
+                isLowering = true;
+                compilation = localRewriter._compilation;
+                inExpressionLambda = localRewriter._inExpressionLambda;
+                currentMethod = localRewriter._factory.CurrentMethod;
+                diagnostics = localRewriter._diagnostics;
```

Comment:
>diagnostics = localRewriter._diagnostics [](start = 16, length = 40)

It looks like there is no good reason to pass diagnostics in as a parameter, we can simply create new bag in the else portion.